### DISCUSSION TITLE
WSA version 2211 with Android 13 now available!

### DISCUSSION THREAD

**Discussion by timkur** [microsoft/WSA#167] [Announcements]
Today we are shipping an update for Windows Subsystem for Android™ on Windows 11 to members of the [Windows Subsystem for Android™ Preview Program](https://learn.microsoft.com/en-us/windows/android/wsa/preview-program). This update (2211.40000.7.0) will update WSA to Android 13 and includes general reliability and performance fixes.  

 
**What’s New**  

- Windows subsystem for Android updated to Android 13 
- Added a new command that shuts down WSA for automation 
- Improvements in boot performance (50%, P10 case)
- Improvements to mouse click input
- Improvements in clipboard stability 
- Improvements to application resizing 
- Upgraded to Intel bridge technology for Android 13 
- Reliability improvements to media files opening in Windows 
- Jumplist entries for applications supporting app shortcuts 

**Giving feedback**  
  
If you are having issues with Windows Subsystem for Android™ – please file feedback via Feedback Hub under Apps > Windows Subsystem for Android™. For more information about troubleshooting and submitting feedback: [Troubleshooting and FAQ for mobile apps on Windows](https://support.microsoft.com/en-us/windows/troubleshooting-and-faq-for-mobile-apps-on-windows-d6062afd-98a1-4018-a7c8-6b3b680a2ea5). 

Thanks,  
Windows Subsystem for Android™ Team

**Comment by ghost** [comment]
Thanks, surprised it wasn't posted as usual in the blog where I keep tabs on: https://blogs.windows.com/blog/tag/windows-insider-program/ 


Also, what does **P10 case** mean in "Improvements in boot performance (50%, P10 case)" ?

**Reply by timkur** [reply]
We only published this to our WSA Preview Program, and not to Windows Insiders, as we're still getting valuable data on 2210 as we prepare to release that to the general audience. Hence, we didn't publish this to the WIP blog.

When we talk about performance, we sometimes talk about it in the P10/P50/P90 case. You can do some internet research on what that means, but here, with noncompeting workloads on a relatively performant device (no spinning disks) we've cut boot time by about half.

Thanks!

Tim

**Reply by rp1231** [reply]
Is there any place to get the WSA Preview Program 2211 msix file without being enrolled in the preview program?

**Reply by ghost** [reply]
Thank you, that's great to know.
I did research about it, now I understand it's Swanson’s probability rule of 30–40–30, from what I've read so far, P10 and P90 correspond to the values in the distribution that aren't as common as P50.

**Reply by timkur** [reply]
> Is there any place to get the WSA Preview Program 2210 msix file without being enrolled in the preview program?

I'm not aware of any official means to get 2210 without being in WIP, nor a way to get 2211 without being in the WSA preview program.

Thanks!

Tim

**Comment by s1204IT** [comment]
[tweet](https://twitter.com/s1204IT/status/1601404376182296576)

We want the **Google Play Store** rather than the **Amazon Appstore**.
Also, [Google Play Games](https://play.google.com/googleplaygames) that can run on Windows is impractical.

Please implement GMS **as an official measure**.

Nothing is less in demand than the **Amazon Appstore**.

**Reply by s1204IT** [reply]
https://github.com/microsoft/WSA/issues/74
We know this is a big challenge for Microsoft.
But **that's** what we really want.

**Reply by rp1231** [reply]
@s1204IT Where did you get the 2211 msix?

**Reply by s1204IT** [reply]
Downloaded as WIP directly from MS Store.
But I couldn't catch MsixBundle in Proxy.

However, it is possible to make an Msix.
Copy all the files from the installation directory and compress them as a ZIP.
Save it with the following file name:
```
WsaPackage_{VER}_{ARCH}_Release-{RING}.msix
```
Example:
```
WsaPackage_2211.40000.7.0_x64_Release-Nightly.msix
```

**Reply by infinitepower18** [reply]
I agree that the Amazon Appstore is useless but I don't know if Google is willing to partner with MS on this. They cannot include any workarounds as they will definitely get into trouble.

**Reply by KHeresy** [reply]
Please provide Google Play system.

**Reply by GuyInCanada** [reply]
> @s1204IT Where did you get the 2211 msix?

here's a link for it.

MicrosoftCorporationII.WindowsSubsystemForAndroid_2211.40000.10.0_neutral_~_8wekyb3d8bbwe

http://tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/6b10fc29-a607-47c1-98b6-8808d7839656?P1=1671293007&P2=404&P3=2&P4=QsXvndEyxKGDuZP6DJrF9%2fXOd7aVyxGE9dE6P8edi4Gn6GW%2fxd41u4LEhMhEA1yzZnMx34ofghLJoVbyHbvX0g%3d%3d

**Reply by s1204IT** [reply]
> > @s1204IT Where did you get the 2211 msix?
> 
> here's a link for it.
> 
> MicrosoftCorporationII.WindowsSubsystemForAndroid_2211.40000.10.0_neutral_~_8wekyb3d8bbwe
> 
> http://tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/6b10fc29-a607-47c1-98b6-8808d7839656?P1=1671257454&P2=404&P3=2&P4=Ox5JqztQKyzNpjaGlEmedrv8EFk5nzagW7k5x7clz3YFu6QC2%2fopSK4tt9g7nK8kISm%2bRDOLw%2f4SNWptO%2fjUcw%3d%3

Thanks,
I'm downloading now.

**Reply by s1204IT** [reply]
@GuyInCanada
As a thank you, I will share the patched file of Mag\*skOnWSA.

https://www.mediafire.com/file/obddq4979ucw2zn/WSA_2211.40000.10.0_x64_Release-Nightly-with-magisk-a468fd94(25205)-canary-MindTheGapps-13.0-RemovedAmazon.7z/file

```
AlternateForm Qualifiers: LIGHTUNPLATED,UNPLATED
Language Qualifiers: EN-US,EN-GB,AF-ZA,AR-SA,AZ-LATN-AZ,BG-BG,BS-LATN-BA,CA-ES,CS-CZ,CY-GB,DA-DK,DE-DE,EL-GR,ES-ES,ES-MX,ET-EE,EU-ES,FA-IR,FI-FI,FR-CA,FR-FR,GL-ES,HE-IL,HI-IN,HR-HR,HU-HU,ID-ID,IS-IS,IT-IT,JA-JP,KA-GE,KK-KZ,KO-KR,LT-LT,LV-LV,MS-MY,NB-NO,NL-NL,NN-NO,PL-PL,PT-BR,PT-PT,QPS-PLOC,QPS-PLOCA,QPS-PLOCM,RO-RO,RU-RU,SK-SK,SL-SI,SQ-AL,SR-CYRL-RS,SR-LATN-RS,SV-SE,TH-TH,TR-TR,UK-UA,VI-VN,ZH-CN,ZH-TW
Scale Qualifiers: 200,400,150,125,100
TargetSize Qualifiers: 256,16,20,24,30,32,36,40,48,60,64,72,80,96
```

**Reply by rp1231** [reply]
@GuyInCanada  Thanks for the link, but it is showing access denied for me.

@s1204IT Would it be possible for you to post a version with OpenGapps instead of mindTheGapps?
Thanks

**Reply by s1204IT** [reply]
@rp1231

Download here:
https://www.mediafire.com/file/oy00ykolrjy4f6n/WSA_2211.40000.10.0_x64_Release-Nightly-with-magisk-a468fd94(25205)-canary-OpenGApps-11.0-pico-RemovedAmazon.7z/file
※Operation has not been confirmed.

Direct link(s) to `MsixBundle` are temporarily accessible.

**Reply by rp1231** [reply]
@s1204IT Thanks a lot!

**Reply by ghost** [reply]
These modified unofficial pre-built files can be malicious since they're Not from a trusted source and we don't know what it contains exactly.

**Reply by GuyInCanada** [reply]
> These modified unofficial pre-built files can be malicious since they're Not from a trusted source and we don't know what it contains exactly.

the scripts used to create them are available on GitHub, if they were malicious, they wouldn't be on GitHub.

WSAGAScript, MagiskOnWSALocal, MagiskOnWSA

**Reply by ghost** [reply]
I know the scripts are on Github, but you missed the point. they are pre-built files, we Don't know if those scripts were used to build them and there wasn't any additional modifications made to them. also one shouldn't trust someone who hosts modified pre-built files on mediafire. always best to use the scripts yourself to build the WSA files.

also, malicious is a broad word, files on Github can be malicious, meaning they can have unwanted/unexpected behaviors, such as calling home or sending data to a 3rd party.

Those are the facts, but if you trust some random person on the Internet claiming they have 100% safe files, go ahead, use them, I myself will only use trusted sources.

**Reply by s1204IT** [reply]
It is up to the user to decide whether to actually use it or not.
Users who don't understand the risks won't ask for files here in the first place.

**Reply by infinitepower18** [reply]
The only reason why Google Play Games exists is because Google is unwilling to partner with MS to include Google Play on WSA. Otherwise what's the point of the Google Play Games program?

Google historically has never supported MS in any of their initiatives. This is the same company that tried to destroy Windows Phone by repeatedly killing the YouTube app that MS made and never developed any apps for the platform. They have zero presence on the Microsoft Store, even Apple has their apps on the MS Store. I heard of them doing shady things with the Edge browser as well.

I would love to be proven wrong and maybe one day Google Play may be integrated into WSA. But chances of this are very low. Developers who develop for WSA are making the apps with Amazon Appstore in mind and changing this suddenly would be a problem.

**Reply by s1204IT** [reply]
I sympathize with your(@infinitepower18) comment.
However, there is no change in the fact that most users want WSA to integrate GMS.

If MS can't make changes directly to WSA, it shouldn't block any means of indirectly integrating GMS.

**Reply by ghost** [reply]
> Users who don't understand the risks won't ask for files here in the first place.

Lol if only that was true...we can only wish. but it's far from the truth.

**Reply by ghost** [reply]
> The only reason why Google Play Games exists is because Google is unwilling to partner with MS to include Google Play on WSA. Otherwise what's the point of the Google Play Games program?
> 
> Google historically has never supported MS in any of their initiatives. This is the same company that tried to destroy Windows Phone by repeatedly killing the YouTube app that MS made and never developed any apps for the platform. They have zero presence on the Microsoft Store, even Apple has their apps on the MS Store. I heard of them doing shady things with the Edge browser as well.
> 
> I would love to be proven wrong and maybe one day Google Play may be integrated into WSA. But chances of this are very low. Developers who develop for WSA are making the apps with Amazon Appstore in mind and changing this suddenly would be a problem.

Totally agree, so considering that, Microsoft should at the very least make it much easier for end-users to root the WSA, preferably with 1 -click, and easily install Gapps or MicroG etc.

Microsoft doesn't need to officially provide the binaries, just make the WSA compatible by default to use Gapps or similar things.

currently, we need to download WSA msix, then use Linux subsystem to modify it, root it etc. and then install Gapps, Microsoft can automate those extra steps for us.

what do you think?

**Reply by gbraad** [reply]
Microsoft won't provide this due to agreements with partners. 
Share the steps and make it into a powershell or batch.

**Reply by ghost** [reply]
> Microsoft won't provide this due to agreements with partners. Share the steps and make it into a powershell or batch.

Huh? which agreement prevents Microsoft to make it easier for end users to root WSA? and which partners are we talking about? Amazon?

**Comment by infinitepower18** [comment]
What's the command to shut down WSA?

**Reply by jaholme** [reply]
wsaclient.exe /shutdown
